config/routes.rb

```diff
@@ -123,7 +123,12 @@
     resource :search,    only: :show
     resource :dashboard, only: :show, constraints: { format: /html|atom/ }
     resources :profiles, only: :show
-    resource :profile, only: [:edit, :update]
+    resource :profile, only: [:edit, :update] do
+      member do
+        get :delete
+        delete :destroy, as: :destroy
```

Comment:
Not really. Of course thats the first thing I tried. Following is what it generates:
```
edit_profile_path	GET	/profile/edit(.:format)	profiles#edit {:format=>:html}
                    PATCH	/profile(.:format)	profiles#update {:format=>:html}
                    PUT	/profile(.:format)	profiles#update {:format=>:html}
                    DELETE	/profile(.:format)	profiles#destroy {:format=>:html}
```
So the path name is `edit_profile_path` with method: delete, right? Except it is not. `DELETE	/profile(.:format)` is unnamed and `edit_profile_path` == `/profile/edit`. Thats why I had to use it as `member`.